sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -747,19 +605,20 @@ private string GetRemappedName(string name)
             return remappedName;
         }
 
-        private string GetRemappedTypeName(NamedDecl namedDecl, Type type)
+        private string GetRemappedTypeName(NamedDecl namedDecl, Type type, out string nativeTypeName)
         {
-            var name = GetTypeName(namedDecl, type);
+            var name = GetTypeName(namedDecl, type, out nativeTypeName);
             return GetRemappedName(name);
         }
 
-        private string GetTypeName(NamedDecl namedDecl, Type type)
+        private string GetTypeName(NamedDecl namedDecl, Type type, out string nativeTypeName)
         {
             var name = type.Spelling;
+            nativeTypeName = name;
 
             if (type is ArrayType arrayType)
             {
-                name = GetTypeName(namedDecl, arrayType.ElementType);
+                name = GetTypeName(namedDecl, arrayType.ElementType, out var nativeElementTypeName);
```

Comment:
The secondary variables here are only used for debugging purposes. In some future PR, they might be used to drop more `NativeTypeNameAttributes` when possible (but the logic for treating `unsigned int *` as equal to `uint*` or transforming `const unsigned int * const *` to be `const uint * const *` is a bit tricky).